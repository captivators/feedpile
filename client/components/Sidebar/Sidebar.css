.sidebar-container {
  position: fixed;
  min-width: 250px;
  width: 250px;
  height: 100vh;
  border-right: 1px solid #888;
  background-color: #f2f2f2;
  z-index: 1001;
}

.sidebar-logo {
  height: 80px;
  width: auto;
  display: block;
  margin: auto;
  padding-top: 50px;
}

.logo-name {
  font-family: 'Bitter', serif;
  font-weight: 700;
  font-size: 28px;
  text-align: center;
  margin-top: 10px;
  margin-bottom: 30px;
}

.refresh-icon {
  position: absolute;
  bottom: 0px;
  left: 0px;
}

.archive-icon {
  position: absolute;
  bottom: 0px;
  right: 0px;
}

@media (max-width: 600px) {
  .sidebar-container {
    display: none;
  }
}
