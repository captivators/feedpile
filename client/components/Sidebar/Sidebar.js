import React, {Component} from 'react';
import {List, ListItem, makeSelectable} from 'material-ui/List';
import PropTypes from 'prop-types';
import AddFeed from '../AddFeed/AddFeed'
import ListIcon from 'material-ui-icons/List';
import LaptopMac from 'material-ui-icons/LaptopMac';
import Star from 'material-ui-icons/Star';
import Games from 'material-ui-icons/Games';
import MusicNote from 'material-ui-icons/MusicNote';
import Public from 'material-ui-icons/Public';
import Refresh from 'material-ui-icons/Refresh';
import Archive from 'material-ui-icons/Archive';
import AddCircle from 'material-ui-icons/AddCircle';
import IconButton from 'material-ui/IconButton';
import Description from 'material-ui-icons/Description';
import feedPileImg from '../../images/feedpile.png'

import './Sidebar.css';
import { getArticlesFromDb, toggleListItem, toggleModal } from '../../actions';
import { connect } from 'react-redux';

const styles = {
  smallIcon: {
    width: 36,
    height: 36
  },
  small: {
    width: 72,
    height: 72,
    padding: 16
  }
};

let SelectableList = makeSelectable(List);


function wrapState(ComposedComponent) {
  return class SelectableList extends Component {
    static propTypes = {
      children: PropTypes.node.isRequired,
      defaultValue: PropTypes.number.isRequired,
    };

    componentWillMount() {
      this.setState({
        selectedIndex: this.props.defaultValue,
      });
    }

    handleRequestChange = (event, index) => {
      this.setState({
        selectedIndex: index,
      });
    };

    render() {
      return (
        <ComposedComponent
          value={this.state.selectedIndex}
          onChange={this.handleRequestChange}
        >
          {this.props.children}
        </ComposedComponent>
      );
    }
  };
}

SelectableList = wrapState(SelectableList);




const Sidebar = (props) => {
  return (
    <div className="sidebar-container">
      <img className="sidebar-logo" src={feedPileImg} />
      <h2 className="logo-name">FeedPile</h2>
      <SelectableList defaultValue={1}>
        <ListItem value={1} primaryText="All Articles" leftIcon={<ListIcon />} />
        <ListItem value={2} primaryText="Starred" leftIcon={<Star />} />
        <ListItem
          value={4}
          primaryText="Technology"
          leftIcon={<LaptopMac />}
          initiallyOpen={props.open}
          onNestedListToggle={props.dispatchToggle}
          primaryTogglesNestedList={false}
          nestedItems={[
            <ListItem
              value={5}
              key={1}
              primaryText="TechCrunch"
              leftIcon={<Description />}
            />,
            <ListItem
              value={6}
              key={2}
              primaryText="The Verge"
              leftIcon={<Description />}
            />,
            <ListItem
              value={7}
              key={3}
              primaryText="WIRED"
              leftIcon={<Description />}
            />
          ]}
        />
        <ListItem
          value={8}
          primaryText="Music"
          leftIcon={<MusicNote />}
          initiallyOpen={false}
          primaryTogglesNestedList={false}
          onNestedListToggle={props.dispatchToggle}
          nestedItems={[
            <ListItem
              value={9}
              key={1}
              primaryText="SPIN"
              leftIcon={<Description />}
            />,
            <ListItem
              value={10}
              key={2}
              primaryText="Rolling Stone"
              leftIcon={<Description />}
            />,
            <ListItem
              value={11}
              key={3}
              primaryText="Pitchfork"
              leftIcon={<Description />}
            />
          ]}
        />
        <ListItem
          value={12}
          primaryText="Games"
          leftIcon={<Games />}
          initiallyOpen={false}
          primaryTogglesNestedList={false}
          onNestedListToggle={props.dispatchToggle}
          nestedItems={[
            <ListItem
              value={13}
              key={1}
              primaryText="IGN"
              leftIcon={<Description />}
            />,
            <ListItem
              value={14}
              key={2}
              primaryText="Gamespot"
              leftIcon={<Description />}
            />,
            <ListItem
              value={15}
              key={3}
              primaryText="Rock, Paper, Shotgun"
              leftIcon={<Description />}
            />
          ]}
        />
        <ListItem
          value={16}
          primaryText="World News"
          leftIcon={<Public />}
          initiallyOpen={false}
          primaryTogglesNestedList={false}
          nestedItems={[
            <ListItem
              value={17}
              key={1}
              primaryText="CNN"
              leftIcon={<Description />}
            />,
            <ListItem
              value={18}
              key={2}
              primaryText="BBC"
              leftIcon={<Description />}
            />,
            <ListItem
              value={19}
              key={3}
              primaryText="NY Times"
              leftIcon={<Description />}
            />
          ]}
        />
        <ListItem value={20} primaryText="Archived" leftIcon={<Archive />} />
      </SelectableList>

      <span className="refresh-icon">
        <IconButton onClick={props.getArticlesFromDb} className="refresh-icon" iconStyle={styles.smallIcon} style={styles.small}>
          <Refresh />
        </IconButton>
      </span>
      <span className="archive-icon">
        <IconButton onClick={() => {props.toggleModal(true)}} iconStyle={styles.smallIcon} className="add-icon" style={styles.small}>
          <AddCircle />
        </IconButton>
      </span>
      <span className="archive-icon">
        <IconButton iconStyle={styles.smallIcon} className="archive-icon" style={styles.small}>
          <Archive />
        </IconButton>
      </span>
      <AddFeed />
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    open: state.open
  }
};

export const Unwrapped = Sidebar;

export default connect(mapStateToProps, {
  dispatchToggle : toggleListItem, getArticlesFromDb, toggleModal
})(Sidebar);
