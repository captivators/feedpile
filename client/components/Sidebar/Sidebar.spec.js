import React from 'react';
import Sidebar, {Unwrapped as UnwrappedSidebar} from './Sidebar';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import injectTapEventPlugin from 'react-tap-event-plugin';
injectTapEventPlugin();

test('Sidebar snapshot test', () => {
  const props = {
    open: true,
    dispatchToggle: () => {}
  };
  const component = shallow(<UnwrappedSidebar {...props}/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

// test('Sidebar renders SidebarAll', () => {
//   const props = {
//     open: true,
//     dispatchToggle: () => {}
//   };
//   const component = mount(<MuiThemeProvider><UnwrappedSidebar {...props}/></MuiThemeProvider>);
//   expect(component.find('ListItem').length).toEqual(9); //number of ListItem's in Sidebar
// });
