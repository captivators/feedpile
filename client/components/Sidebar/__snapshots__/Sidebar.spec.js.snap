// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar snapshot test 1`] = `
<div
  className="sidebar-container"
>
  <img
    className="sidebar-logo"
    src="https://s3-us-west-1.amazonaws.com/thesis.feedpile.images/feedpile.png"
  />
  <h2
    className="logo-name"
  >
    FeedPile
  </h2>
  <List>
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(List) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="All Articles"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(Star) />}
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Starred"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={true}
      insetChildren={false}
      leftIcon={<pure(LaptopMac) />}
      nestedItems={
        Array [
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="TechCrunch"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="The Verge"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="WIRED"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Technology"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(MusicNote) />}
      nestedItems={
        Array [
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="SPIN"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Rolling Stone"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Pitchfork"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Music"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(Games) />}
      nestedItems={
        Array [
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="IGN"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Gamespot"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Rock, Paper, Shotgun"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Games"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={<pure(Public) />}
      nestedItems={
        Array [
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="CNN"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="BBC"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
          <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftIcon={<pure(Description) />}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="NY Times"
            primaryTogglesNestedList={false}
            secondaryTextLines={1}
        />,
        ]
      }
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="World News"
      primaryTogglesNestedList={true}
      secondaryTextLines={1}
    />
  </List>
  <span
    className="refresh-icon"
  >
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "height": 36,
          "width": 36,
        }
      }
      onClick={[Function]}
      style={
        Object {
          "height": 72,
          "padding": 16,
          "width": 72,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}
    >
      <pure(Refresh) />
    </IconButton>
  </span>
  <span
    className="archive-icon"
  >
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "height": 36,
          "width": 36,
        }
      }
      style={
        Object {
          "height": 72,
          "padding": 16,
          "width": 72,
        }
      }
      tooltipPosition="bottom-center"
      touch={false}
    >
      <pure(Archive) />
    </IconButton>
  </span>
</div>
`;
